api/plugins.js

```diff
@@ -3,33 +3,44 @@ import { deferOrRunWrap, successStopOrErrorWrap } from '../lib/lifecycle.js'
 export function removePlugin (description, config) {
   return deferOrRunWrap(function (context) {
     return successStopOrErrorWrap('removePlugin', description, async () => {
-      context.journal.freeze()
 
-      context.journal.unfreeze()
-      return result
+      if (!description || !config) throw new Error('removePlugin - incorrectly configured')
+
+      context.fromPlugins = context.fromPlugins.filter(plugin => plugin.name !== config.name)
+
+      return true
     })
   }, { description, type: 'action' })
 };
 
 export function addPlugin (description, config) {
   return deferOrRunWrap(function (context) {
     return successStopOrErrorWrap('addPlugin', description, async () => {
-      context.journal.freeze()
 
-      context.journal.unfreeze()
-      return result
+      if (!description || !config) throw new Error('addPlugin - incorrectly configured')
+
+      const newPlugin = context.toPlugins.filter(plugin => (plugin.name === config.name || plugin.displayName?.toUpperCase() === config.name?.toUpperCase() ))
+      if (newPlugin.length === 0) throw new Error(`addPlugin - ${config.name} not found`)
+      context.fromPlugins.push(newPlugin[0])
```

Comment:
This is a bit clearer now.
```suggestion
      const newPlugin = context.toPlugins.find(plugin => (plugin.name === config.name || plugin.displayName?.toUpperCase() === config.name?.toUpperCase() ))
      if (!newPlugin) throw new Error(`addPlugin - ${config.name} not found`)
      context.fromPlugins.push(newPlugin)
```

Array.find returns the first found matching item.

Reference:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
@oliverfoster I was unsure whether the toUpperCase was needed, obviously prevents incorrect capitalisation being an issue, but not sure if it's just unnecessary? If you prefer leaving it in I can add it to the other functions as well, was more of a use case.

Will change to find.